rcljava/src/main/java/org/ros2/rcljava/action/ActionServerImpl.java

```diff
@@ -432,6 +507,36 @@ private static native boolean nativeCheckGoalExists(
     long goalInfoFromJavaConverterHandle,
     long goalInfoDestructorHandle);
 
+  private boolean goalExists(action_msgs.msg.GoalInfo goalInfo) {
+    long goalInfoFromJavaConverterHandle = goalInfo.getFromJavaConverterInstance();
+    long goalInfoDestructorHandle = goalInfo.getDestructorInstance();
+
+    return nativeCheckGoalExists(
+      this.handle, goalInfo, goalInfoFromJavaConverterHandle, goalInfoDestructorHandle);
+  }
+
+  private action_msgs.msg.GoalInfo createGoalInfo(List<Byte> goalUuid) {
+    // TODO(ivanpauno): Do we always need to set the time here?
```

Comment:
I think we only record the time when the goal was accepted in case the user wants to query it. It looks like we don't need the timestamp in the case where we re-use this method to check for the existence of a goal (at L638). Maybe setting the timestamp could be moved just after calling this method at L371. I don't think it's causing much harm leaving it here though either.

Reply:
I think it isn't needed at L371 either, see https://github.com/ros2/rcl/blob/414fdbd479b83f2b8af8235ad1fa3aeba2b0027f/rcl_action/src/rcl_action/action_server.c#L382-L389.
In rclcpp, the timestamp got in the goal request is passed through. I could also extend the `GoalRequestDefinition` interface to be able to get the timestamp, but I don't think that's needed.

Removed the code setting the timestamp in https://github.com/osrf/ros2_java/pull/65/commits/88346c2d2ad1161d2ad780e94d2f3b6571508e90.